### PR TITLE
 Add select all / select none on the IGV file selector, enforce max number of selected files

### PR DESCRIPTION
**Pull request by wnojopra**
Screenshot from 2019-05-02 13-13-13

### CODE REVIEW

File: src/components/IGVFileSelector.js [hunk 1]
```diff
@@ -49,16 +72,19 @@ export class IGVFileSelector extends Component {
...
-            _.keys,
-            _.filter(v => selectedFiles[v])
-          )(selectedFiles)
-          onSuccess(actuallySelectedFiles)
+          onSuccess(this.getSelectedFilesList())
         }
       }, ['Done'])
     }, [
+      div({ style: { marginBottom: '1rem', display: 'flex' } }, [
+        div({ style: { fontWeight: 500 } }, ['Show:']),
```
Comment (panentheos) [inline src/components/IGVFileSelector.js]: How about 'Select' instead of 'Show' here?

File: src/components/IGVFileSelector.js [hunk 2]
```diff
@@ -41,6 +43,27 @@ export class IGVFileSelector extends Component {
...
+  }
+
+  buttonIsDisabled() {
+    const selectedFilesList = this.getSelectedFilesList()
+    return !(selectedFilesList.length > 0 && selectedFilesList.length <= MAX_CONCURRENT_IGV_FILES)
+  }
+
+  getButtonTooltip() {
+    return this.buttonIsDisabled() ? `Select between 1 and ${MAX_CONCURRENT_IGV_FILES} files` : ''
+  }
```
Comment (panentheos) [inline src/components/IGVFileSelector.js]: nit: These calculations could be inlined into the render method, which would likely improve readability and cut down on extraneous computation a bit.

**Review by panentheos — APPROVED** [review]
Looks good, see a few minor comments.

**Comment by wnojopra** [comment]
Thanks @panentheos

File: src/components/IGVFileSelector.js [hunk 3]
```diff
@@ -22,10 +22,12 @@ const getStrings = v => {
   )
 }
 
+const MAX_CONCURRENT_IGV_FILES = 10
```
Comment (cahrens) [inline src/components/IGVFileSelector.js:25]: @wnojopra do you recall why you limited the number of files that could be opened at a time (the limit was just removed per https://github.com/DataBiosphere/terra-ui/pull/2943)?

Comment (wnojopra) [reply src/components/IGVFileSelector.js:25]: The details are unfortunately lost in auto-deleted emails, but I recall it was a product decision. I believe there was feedback where if you had a lot of files, you would be easily tempted to open them all up at once. I don't remember the exact behavior of what happens if you do this, but it was fairly unusable, and you ended up downloading a lot of data for nothing.

I don't use IGV at all these days, but there might be some reasonable number for a limit. Perhaps 10 is too low, but I wouldn't want to enable a state where we download a lot of data but end up in an unusable state. 

Or perhaps IGV behaves differently today than it did three years ago, and we don't need a limit anymore!

**Pull request merged**